packages/loot-core/src/types/server-handlers.d.ts

```diff
@@ -283,7 +283,8 @@ export interface ServerHandlers {
   'subscribe-needs-bootstrap': (
     args: { url } = {},
   ) => Promise<
-    { error: string } | { bootstrapped: unknown; hasServer: boolean }
+    | { error: string }
+    | { bootstrapped: unknown; hasServer: boolean; error?: undefined }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I don't think we need `error` here
```suggestion
    | { bootstrapped: unknown; hasServer: boolean; }
```

Reply:
@MatissJanis 
We currently have a deconstruction of the return of this handler here: https://github.com/actualbudget/actual/blob/d8996405c43a8b1dd530ccceeb78fe049e03919e/packages/loot-core/src/server/main.ts#L1673

typescript won't let you use deconstruction on an object with union types unless the properties you are deconstructing exist on all types in the union https://github.com/microsoft/TypeScript/issues/12815

wasn't sure whether we wanted to drop the deconstruction in favour of a result object, or adjust the typescript to allow us to keep the deconstruction. Thoughts?

Reply:
The ultimate source of truth should be [the handler](https://github.com/actualbudget/actual/blob/d8996405c43a8b1dd530ccceeb78fe049e03919e/packages/loot-core/src/server/main.ts#L1519-L1521) itself. Which never returns `error` together with the other properties.

So in that sense: the consumer (destruction) is wrong and should be changed. Would you be open to changing it slightly?